Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/parser/TestPatientDiscoveryMessageHelper.java

```diff
@@ -0,0 +1,322 @@
+            bday.setValue(birthTime);
+            subjectBirthTime.getValue().add(bday);
+        }
+
+        return subjectBirthTime;
+    }
+
+    private static PRPAMT201306UV02LivingSubjectAdministrativeGender createGender(String gender) {
+        PRPAMT201306UV02LivingSubjectAdministrativeGender adminGender
+            = new PRPAMT201306UV02LivingSubjectAdministrativeGender();
+
+        if (gender != null && gender.length() > 0) {
+            CE genderCode = new CE();
+            genderCode.setCode(gender);
+            adminGender.getValue().add(genderCode);
+        }
+
+        return adminGender;
+    }
+
+    private static PRPAIN201306UV02MFMIMT700711UV01ControlActProcess createResponseControlActProcess() {
+        PRPAIN201305UV02 query = createPRPAIN201305UV02Request();
+
+        Patients patients = createPatients();
+
+        PRPAIN201306UV02MFMIMT700711UV01ControlActProcess controlActProcess
+            = new PRPAIN201306UV02MFMIMT700711UV01ControlActProcess();
+
+        controlActProcess.setMoodCode(XActMoodIntentEvent.EVN);
+        controlActProcess.setClassCode(ActClassControlAct.CACT);
+
+        CD code = new CD();
+        code.setCode("PRPA_TE201306UV02");
+        code.setCodeSystem("2.16.840.1.113883.1.6");
+        controlActProcess.setCode(code);
+
+        if (patients != null && !patients.isEmpty()) {
+            for (Patient patientEntry : patients) {
+                controlActProcess.getSubject().add(createSubject(patientEntry, query));
+            }
+        }
+
+        // Add in query parameters
+        if (query.getControlActProcess() != null && query.getControlActProcess().getQueryByParameter() != null
+            && query.getControlActProcess().getQueryByParameter().getValue() != null) {
+            controlActProcess.setQueryByParameter(query.getControlActProcess().getQueryByParameter());
+        }
+        // Set original QueryByParameter in response
+        controlActProcess.setQueryByParameter(query.getControlActProcess().getQueryByParameter());
+        return controlActProcess;
+    }
+
+    private static PRPAIN201306UV02MFMIMT700711UV01Subject1 createSubject(Patient patient, PRPAIN201305UV02 query) {
+        PRPAIN201306UV02MFMIMT700711UV01Subject1 subject = new PRPAIN201306UV02MFMIMT700711UV01Subject1();
+        subject.getTypeCode().add("SUBJ");
+        subject.setRegistrationEvent(createRegEvent(patient, query));
+
+        return subject;
+    }
+
+    private static PRPAIN201306UV02MFMIMT700711UV01RegistrationEvent createRegEvent(Patient patient,
+        PRPAIN201305UV02 query) {
+
+        II id = new II();
+        id.getNullFlavor().add("NA");
+
+        CS statusCode = new CS();
+        statusCode.setCode("active");
+
+        PRPAIN201306UV02MFMIMT700711UV01RegistrationEvent regEvent
+            = new PRPAIN201306UV02MFMIMT700711UV01RegistrationEvent();
+
+        regEvent.getMoodCode().add("EVN");
+        regEvent.getClassCode().add("REG");
```

Comment:
Above, you use Mood and Class code constants, here you use an in-line literal.  Is this intentional?